### ISSUE TITLE
Can't assign I18NexFsBackend to NewableModule<BackendModule>

### ISSUE THREAD

**Issue by miniplus** [i18next/i18next-fs-backend#32]
### Error:
`Property 'type' is missing in type 'typeof I18NexFsBackend' but required in type 'NewableModule<BackendModule<unknown>>'.`

### Debug info
```ts
export interface NewableModule<T extends Module> extends Newable<T> {
    type: T['type'];
}
```

Essentially becomes:
```ts
export interface NewableModule<T extends Module> {
    type: T['type'];
    new (...args: any[]): T
}
```

Which means the `type` property is expected to be static.

After investigation I noticed the now legacy [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/i18next-fs-backend/index.d.ts#L38) definitions has both static as well as non-static props defined as well.

I'm guessing their use case was the same.

## To Reproduce

```ts
import type {NewableModule, BackendModule} from "i18next";
import I18NexFsBackend from "i18next-fs-backend";

const backend: NewableModule<BackendModule> = I18NexFsBackend
        ^ Property 'type' is missing in type 'typeof I18NexFsBackend' but required in type 'NewableModule<BackendModule<unknown>>'.
```

## Expected behavior

`I18NexFsBackend` can be assigned to `backend`.

```ts
import type {NewableModule, BackendModule} from "i18next";
import I18NexFsBackend from "i18next-fs-backend";

const backend: NewableModule<BackendModule> = I18NexFsBackend
```

## Your Environment

- *runtime version*: node 16.18.0
- *i18next version*: ^22.0.6
- *os*: Mac
- *typescript*: ^4.6.4

**Comment by adrai** [comment]
Would you like to send a Pull Request to address this?

**Comment by miniplus** [comment]
> Would you like to send a Pull Request to address this?

Was literally in the process. I noticed the same issue for http-backend and I'm guessing most likely other backend as well than.

**Comment by adrai** [comment]
solved with v2.0.1
